src/azul/service/responseobjects/storage_service.py

```diff
@@ -1,34 +1,38 @@
+from concurrent.futures import ThreadPoolExecutor, as_completed
+from dataclasses import dataclass
 from logging import getLogger
-from typing import Optional
+from typing import Optional, List
 import boto3
 from azul import config
 
 logger = getLogger(__name__)
+AWS_S3_DEFAULT_MINIMUM_PART_SIZE = 5242880  # 5 MB; see https://docs.aws.amazon.com/AmazonS3/latest/dev/qfacts.html
+MULTIPART_UPLOAD_MAX_WORKERS = 12
 
 
 class StorageService:
 
     def __init__(self):
-        self.__bucket_name = config.s3_bucket
-        self.__client = None  # the default client will be assigned later to allow patching.
+        self.bucket_name = config.s3_bucket
+        self._client = None  # the default client will be assigned later to allow patching.
 
     @property
     def client(self):
-        if not self.__client:
-            self.__client = boto3.client('s3')
-        return self.__client
+        if not self._client:
+            self._client = boto3.client('s3')
+        return self._client
 
     def set_client(self, client):
-        self.__client = client
+        self._client = client
 
-    def get(self, object_key: str):
+    def get(self, object_key: str) -> str:
```

Comment:
Not happy about the asymmetry between get() and put(). The former returns a string (using the default encoding, which I think should instead be explicitly fixed to utf-8) and the latter takes bytes. Unless there is a compelling reason for the asymmetry, I'd rather avoid it. Sorry for not catching that earlier.

Reply:
When I implement this part. The only compelling reason that I can remember is probably because the code would eventually needs to work with the string version of the data, not bytes. However, I can see your concern here and I now think this definitely will confuse other people. I will change this to bytes.